numba/parfors/array_analysis.py

```diff
@@ -85,17 +85,45 @@ def wrap_index(typingctx, idx, size):
     (idx < 0 ? idx + size : idx) because we may have situations
     where idx > size due to the way indices are calculated
     during slice/range analysis.
+
+    Both idx and size have to be Integer types.
+    size should be from the array size vars that array_analysis
+    adds and the bidwidth should match the platform maximum.
```

Comment:
If the bitwidth should match platform max, could this code just `{z,s}ext` to `intp.bitwidth` and or in the signedness on the chosen type?

Reply:
Are you just suggesting changing "unified_bitwidth = max(idx.bitwidth, size.bitwidth)" to "unified_bitwidth = types.intp.bitwidth"?

Reply:
I simplified it a bit.  Appreciate your input.